html/modules/custom/hpc_api/src/Query/EndpointPlanQuery.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+namespace Drupal\hpc_api\Query;
+
+/**
+ * Get plan information.
+ */
+class EndpointPlanQuery extends EndpointQuery {
+
+  /**
+   * Get plan.
+   */
+  public function getPlan($plan_id) {
+    $this->setEndpoint('public/plan/id/' . $plan_id);
+
+    $data = $this->getData();
+
+    $plan = [
+      'id' => $data->id,
+      'name' => $data->planVersion->name,
+      'code' => $data->planVersion->code,
+      'start_date' => $data->planVersion->startDate,
+      'end_date' => $data->planVersion->endDate,
+      'orig_requirements' => $data->origRequirements,
+      'revised_requirements' => $data->revisedRequirements,
+      'categories' => [],
+      'locations' => [],
+    ];
+
+    foreach ($data->categories as $category) {
+      $plan['categories'][] = $category->name;
+    }
+
+    foreach ($data->locations as $location) {
+      $plan['locations'][] = $location->name;
```

Comment:
Same here.

Reply:
It's based on a module from unocha, so we can add more fields